chain/messagesigner/messagesigner.go

```diff
@@ -29,6 +31,19 @@ type MpoolNonceAPI interface {
 	GetActor(context.Context, address.Address, types.TipSetKey) (*types.Actor, error)
 }
 
+type MsgSigner interface {
+	SignMessage(ctx context.Context, msg *types.Message, spec *api.MessageSendSpec, cb func(*types.SignedMessage) error) (*types.SignedMessage, error)
+	GetSignedMessage(ctx context.Context, uuid uuid.UUID) (*types.SignedMessage, error)
+	StoreSignedMessage(ctx context.Context, uuid uuid.UUID, message *types.SignedMessage) error
+	NextNonce(ctx context.Context, addr address.Address) (uint64, error)
+	SaveNonce(ctx context.Context, addr address.Address, nonce uint64) error
+	DstoreKey(addr address.Address) datastore.Key
+	IsLeader(ctx context.Context) bool
+	RaftLeader(ctx context.Context) (peer.ID, error)
+	RedirectToLeader(ctx context.Context, method string, arg interface{}, ret interface{}) (bool, error)
+	GetRaftState(ctx context.Context) (consensus.State, error)
```

Comment:
Same reason as above. Everything consensus related has to be exposed via the MessageSigner
It needs to be on the interface because the FullNodeAPI needs to have this method also

Reply:
We could have an optional dependency on the Consensus struct for those APIs; The MsgSigner interface should only be used to abstract out message-signing related things:

![image](https://user-images.githubusercontent.com/3867941/191434037-a9f10604-8cfb-4640-883b-11c69c192e92.png)

Reply:
Changed dependencies to the above diagram